core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,138 +19,357 @@
+     * testing.
+     * 
+     * Can be changed back to HashMap once https://issues.apache.org/jira/browse/LUCENE-6305 is fixed.
+     */
+    private final Map<String, Float> fieldsAndWeights = new TreeMap<>();
+    /** If specified, analyzer to use to parse the query text, defaults to registered default in toQuery. */
+    private String analyzer;
+    /** Name of the query. Optional.*/
     private String queryName;
+    /** Default operator to use for linking boolean clauses. Defaults to OR according to docs. */
+    public static final Operator DEFAULT_OPERATOR = Operator.OR;
+    private Operator defaultOperator = DEFAULT_OPERATOR;
+    /** If result is a boolean query, minimumShouldMatch parameter to apply. Ignored otherwise. */
     private String minimumShouldMatch;
-    private int flags = -1;
-    private Boolean lowercaseExpandedTerms;
-    private Boolean lenient;
-    private Boolean analyzeWildcard;
-    private Locale locale;
+    /** Any search flags to be used, ALL by default. */
+    public static final int DEFAULT_FLAGS = SimpleQueryStringFlag.ALL.value;
+    private static final int INVALID_FLAGS = -42;
+    private int flags = DEFAULT_FLAGS;
+
+    /** Further search settings. */
+    private Settings settings = new Settings();
+
     static final SimpleQueryStringBuilder PROTOTYPE = new SimpleQueryStringBuilder(null);
 
-    /**
-     * Operators for the default_operator
-     */
+    /** Operators available for linking boolean clauses. */
+    // Move out after #11345 is in.
     public static enum Operator {
         AND,
-        OR
+        OR;
+
+        public static Operator parseFromInt(int ordinal) {
+            switch (ordinal) {
+                case 0:
+                    return AND;
+                case 1:
+                    return OR;
+                default:
+                    throw new IllegalArgumentException("Cannot parse Operator from ordinal " + ordinal);
+            }
+
+        }
     }
 
-    /**
-     * Construct a new simple query with the given text
-     */
-    public SimpleQueryStringBuilder(String text) {
-        this.queryText = text;
+    /** Construct a new simple query with this query string. */
+    public SimpleQueryStringBuilder(String queryText) {
+        this.queryText = queryText;
     }
 
-    /**
-     * Add a field to run the query against
-     */
+    /** Returns the text to parse the query from. */
+    public String text() {
+        return this.queryText;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public SimpleQueryStringBuilder boost(float boost) {
+        this.boost = boost;
+        return this;
+    }
+    
+    /** Returns the boost to apply to resulting Lucene query.*/
+    public float boost() {
+        return this.boost;
+    }
+
+    /** Add a field to run the query against. */
     public SimpleQueryStringBuilder field(String field) {
-        this.fields.put(field, null);
+        if (Strings.isEmpty(field) == true) {
```

Comment:
we don't need `== true`, I think we use this explicit version only for negations, instead of the `!`